include/api/CDataFrameAnalysisConfigReader.h

```diff
@@ -93,6 +93,8 @@ class API_EXPORT CDataFrameAnalysisConfigReader {
         //! Get a boolean parameter.
         bool fallback(bool value) const;
         //! Get an unsigned integer parameter.
+        std::uint64_t fallback(std::uint64_t fallback) const;
+        //! Get an unsigned integer parameter.
```

Comment:
You may find this is problematic on 64 bit Windows, where `long` is still 32 bits, so `size_t` is probably exactly the same underlying type as `uint64_t`, so then this overload will clash with the one immediately below.

I think you should either limit the possible fallback types to just a couple, say `size_t` and `ptrdiff_t`, or else if there are going to be masses of overloads then provide one for every raw type: `short`, `int`, `long`, `long long`, `unsigned short`, `unsigned int`, `unsigned long`, `unsigned long long`.  Then you can call the methods with `size_t`, `uintN_t`, whatever, and the overload mechanism will pick the right overload for the platform.  This is what we had to do in `CStringUtils::typeToString` in the end, and what the standard library equivalents will do.  But if we can get away with just `size_t` and `ptrdiff_t` then that's less verbose.

Reply:
I can get away with `size_t`: I just need to add a cast where it is used to initialise the seed. (Windows CI did compile so I guess the type is different, but I don't actually have strong opinions on this so happy to change.)

Reply:
Actually, I was wrong. Windows didn't compile for this reason, I was looking at the wrong output.